utils/build-script

```diff
@@ -1166,9 +1202,16 @@ details of the setups of other systems or automated environments.""")
 
     host_clang = swift_build_support.toolchain.host_clang(
         xcrun_toolchain=args.darwin_xcrun_toolchain)
-    if not host_clang:
+    if args.host_cc is None and host_clang is not None:
+        args.host_cc = str(host_clang.cc)
+    if args.host_cxx is None and host_clang is not None:
+        args.host_cxx = str(host_clang.cxx)
+    is_host_clang_ok = (
+        args.host_cc is not None and os.access(args.host_cc, os.X_OK) and
+        args.host_cxx is not None and os.access(args.host_cxx, os.X_OK))
```

Comment:
The intent of `Toolchain` object *was* unclear for me. Now, I got it.
But, I think, `Toolchain` should be refactored for that intent.
Hopefully, can I do that in next PR? so as `distcc` thing.

My vague idea is:
`host_toolchain()` must return `Toolchain` object unconditionally.
`toolchain.{toolname}` should dynamically query `xcrun --find` or `which` depends on the platform and the name of the tool, and cache it.
`toolchain.{toolname}` is assignable.